tests/test_prolongation.cpp

```diff
@@ -0,0 +1,257 @@
+#include <gtest/gtest.h>
+#include "gmgpolar.h"
+class test_prolongation : public ::testing::TestWithParam<int>
+{
+protected:
+    void SetUp() override
+    {
+        //initialize default parameters.
+        gyro::init_params();
+        gyro::icntl[Param::verbose]        = 0;
+        gyro::icntl[Param::debug]          = 0;
+        gyro::icntl[Param::extrapolation]  = 0;
+        gyro::icntl[Param::DirBC_Interior] = 1;
```

Comment:
I think we should test with and without Dirichlet conditions

Reply:
Additionally (and in contrast) to my comment about "why do you set it if you don't use it?", we should maybe check with parameters that could make a change as in a future version, these might be implemented here. In particular Dirichlet boundary conditions could lead to a simplification but the operator should still handle both cases correctly. However, some parameters are probably never used in the context of the prolongation (also applies to other functions below)